### PR TITLE
[WIP] [#151] toggle private card writing

### PR DESCRIPTION
**Pull request by BlexBaker**

### CODE REVIEW

File: components/mainContent.html [hunk 1]
```diff
@@ -28,7 +28,7 @@ <h2 ng-show="!editing" ng-click="editing = true; newName=type.value;">
           <li ng-controller="MessageCtrl" id="{{ message.$id }}" enter-click class="message" ng-class="{'flip': isEditing}" ng-repeat="message in messages | filter: {type: {id: type.id}, text: filterMessage} | orderBy: getSortFields()">
             <div messageId="{{ message.$id }}" x-lvl-draggable x-lvl-drop-target="true" x-on-drop="dropCardOnCard(dragEl, dropEl)" draggable="{{ !isEditing }}">
 
-              <div class="front" ng-class="{'filter': message.creating}">
+              <div class="front" ng-class="{'filter': isCensored(message, privateWritingOn)}">
```
Comment (pamo) [inline components/mainContent.html]: nice

File: components/dialogs.html [hunk 2]
```diff
@@ -23,6 +23,10 @@
   <input id="newBoardName" ng-model="newBoard.name" type="text" ng-keyup="submitOnEnter($event, 'createNewBoard')"></input>
   <label for="maxUserVotes">Max votes per user <small>(whole board)</small></label>
   <input id="maxUserVotes" ng-model="newBoard.max_votes" type="number" ng-init="newBoard.max_votes = 6" ng-keyup="submitOnEnter($event, 'createNewBoard')"></input>
+  <div class="labeled-checkbox">
+    <input class="toggle-checkbox" ng-model="newBoard.text_editing_is_private" type="checkbox" ng-init="newBoard.text_editing_is_private = true"></input>
```
Comment (pamo) [inline components/dialogs.html]: `ng-init` is a directive that should be used for ["aliasing special properties of ngRepeat"](https://docs.angularjs.org/api/ng/directive/ngInit). Setting `text_editing_is_private` should be done in the controller, imo.

File: components/dialogs.html [hunk 3]
```diff
@@ -80,3 +84,10 @@
     <span><button class="increment-button" ng-click="voteService.decrementMaxVotes(userId, maxVotes);"><i class="fa fa-minus fa-md decrement-button" aria-hidden="true"></i></button></span>
   </div>
 </script>
+<script type="text/ng-template" id="cardSettings">
+  <h1>Card Settings</h1>
+    <div class="labeled-checkbox">
+      <input class="toggle-checkbox" type="checkbox"  ng-model="privateWritingOn" ng-change="  updatePrivateWritingToggle(privateWritingOn)"></input>
```
Comment (pamo) [inline components/dialogs.html]: You have an extra space in the `ng-change` attribute value.

**Comment by BlexBaker** [comment]
This is what the UI looks like
 
New Board Dialog: 
screen shot 2017-10-17 at 7 48 28 pm

Side Bar: 

screen shot 2017-10-17 at 7 48 38 pm

Settings Dialog: 

screen shot 2017-10-17 at 7 48 44 pm

**Comment by glauberramos** [comment]
PR looks fine, I will just check on localhost and approve it soon.

**Comment by glauberramos** [comment]
There is a small issue that I think we need to fix. There is no listener for changes on the board object, so any client will not update automatically when this setting is toggled.

**Comment by glauberramos** [comment]
opening new pr and closing this

**Pull request closed without merging**